sklearn/utils/_param_validation.py

```diff
@@ -148,6 +156,9 @@ def wrapper(*args, **kwargs):
 class _Constraint(ABC):
     """Base class for the constraint objects."""
 
+    def __init__(self):
+        self.hidden_constraint = False
```

Comment:
Nit: At this point, since it is an attribute on "Constraint", I think we can lose the "_constraint" suffix:

```suggestion
        self.hidden = False
```

Reply:
makes sense